security/src/main/java/uk/ac/ox/ndph/mts/security/authorisation/AuthorisationService.java

```diff
@@ -136,16 +132,54 @@ public boolean authorise(String requiredPermission, List<String> entitiesSiteIds
                 return false;
             }
 
-            List<SiteDTO> sites = siteServiceClient.getAllSites();
+            List<String> userSites = siteServiceClient.getAllSites().stream()
+                    .map(SiteDTO::getSiteId).collect(Collectors.toList());
 
-            return isAuthorisedToAllSites(sites, roleAssignments, entitiesSiteIds);
+            //return true only if authorised to all sites
+            return userSites.containsAll(entitiesSiteIds);
 
         } catch (Exception e) {
             LOGGER.info(String.format("Authorisation process failed. Error message: %s", e.getMessage()));
             return false;
         }
     }
 
+    /**
+     * Filter unauthorised sites
+     * @param sitesReturnObject all sites returned object
+     * @return true if filtering finished successfully
+     */
+    public boolean filterUserSites(List<?> sitesReturnObject) {
+
+        try {
+            Objects.requireNonNull(sitesReturnObject, "sites can not be bull");
+
+            List<SiteDTO> sites = sitesReturnObject.stream()
+                    .map(siteObject -> {
+                        var siteId =  siteUtil.getSiteIdFromObj(siteObject, "getSiteId");
+                        var parentSiteId =  siteUtil.getSiteIdFromObj(siteObject, "getParentSiteId");
+                        return new SiteDTO(siteId, parentSiteId);
+                    }).collect(Collectors.toList());
+
+            String userId = securityContextUtil.getUserId();
+            String token = securityContextUtil.getToken();
+            List<RoleAssignmentDTO> roleAssignments = practitionerServiceClient.getUserRoleAssignments(userId, token);
+            Set<String> userSites = siteUtil.getUserSites(sites, roleAssignments);
+
+            sitesReturnObject.removeIf(siteObject ->
+                    !userSites.contains(siteUtil.getSiteIdFromObj(siteObject, "getSiteId")));
+
+            if (sitesReturnObject.isEmpty()) {
+                return false;
```

Comment:
why 'false'?
to my understasnding false means the filtering failed. but can it be that it was succesfull and the result is an empty list?

Reply:
false in authZ means unauthorised. If the user has no sites that he is authorised to see then the requirement was to deny access